### DISCUSSION TITLE
关于如何进一步提升同时渲染大量字段的性能

### DISCUSSION THREAD

**Discussion by NexxLuo** [alibaba/formily#2170] [Q&A 2.x]
一个很简单的场景，一个表格，存在10列，可能会从后端请求大于100行数据，同时将这100行数据设置给表格，就会产生1000个字段，从给表格setValue到显示出数据这个过程会耗费5s甚至10s的延迟(随字段控件的复杂程度相应增加)，期间甚至会造成浏览器卡死。

这种情况下如何提升性能及用户体验？
排除分页方案、排除虚拟滚动方案。

实例：https://codesandbox.io/s/yi-ci-xing-dong-tai-xuan-ran-da-liang-zi-duan-unkws?file=/App.tsx

**Comment by janryWang** [comment]
哈哈，你这是要硬刚业务变态需求啊

**Comment by janryWang** [comment]
目前我也在思考，应该主要在@formily/path这部分如何提升解析性能着手了，其他的暂时想不到，如果这部分能用个3-4倍的提升，那整个表单性能基本上起飞

**Reply by wsxe9988** [reply]
可以看看vxe-table，有没有什么可以借鉴的

**Reply by janryWang** [reply]
性能不在于组件，在于表单内部的计算
